### DISCUSSION TITLE
inconsistent threading

### DISCUSSION THREAD

**Discussion by yuuahmad** [grblHAL/core#581] [Q&A] [answered]
does anyone here know how to solve this problem? is it using backlash compensation? or does my machine need to be dialed back? or is there a firmware setting that I need to change?

i am using the latest version of grblhal now

my scenario
- my lathe has a backlash of 0.1mm
- threading using G97
- backlash compensation on $160 I turn it off (value is 0)

here I attach a picture and diagram of my problem if there is any incomplete information just write it in the comments column.
![threading dan backlash drawio (1)](https://github.com/user-attachments/assets/e4c8bbe5-c2e8-4b03-90db-13e14aae8a62)
![20240912_092151](https://github.com/user-attachments/assets/9f2b2122-eb7c-41db-9a7e-a64084fd54c7)

**Comment by terjeio** [comment, marked as answer]
Backlash should not be an issue since threading is in one direction?
Which driver/board are you using?
Have you [tuned the PID parameters](https://github.com/grblHAL/core/wiki/Spindle-sync) with good results?
Spindle is relatively stable RPM-vise when threading?
What kind of encoder are you using? Number of of pulses per rev?
